### ISSUE TITLE
How to customise include level?

### ISSUE THREAD

**Issue by lkbr** [jelz/cf-graphql#6]
Is it possible to customise includes to go a level deeper?

I have an `organisation` that references a `service` that references a `category`. Testing out the example with the below did not work:
```javascript
{
  organisations {
    services {
      category {
        name
      }
    } 
  }
}
```

Excited about the potential of using graphql for contentful! Thank you.

**Comment by jelz** [comment]
Hey, I've just tested similar setup and it works just fine. Questions:

- what's the result? Any exceptions logged?
- do you use `npm run example` or your custom server built on top of `cf-graphql`?
- what's the library version you use?

**Comment by lkbr** [comment]
Thanks for the response. I've recloned this repo and run `npm run example`.

The error from GraphiQL:
```
{
  "errors": [
    {
      "message": "Cannot query field \"name\" on type \"Entry\". Did you mean to use an inline fragment on \"OrganisationBranch\", \"Organisation\", \"SubscriptionPlan\", \"Service\", or \"OrganisationCapacityMetrics\"?",
      "locations": [
        {
          "line": 34,
          "column": 5
        }
      ],
      "stack": "GraphQLError\n    at Object.Field (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/validation/rules/FieldsOnCorrectType.js:66:31)\n    at Object.enter (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/language/visitor.js:296:29)\n    at Object.enter (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/language/visitor.js:338:25)\n    at visit (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/language/visitor.js:228:26)\n    at visitUsingRules (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/validation/validate.js:75:22)\n    at validate (/Users/lukebrown/Documents/dev/cf-graphql/node_modules/graphql/validation/validate.js:60:10)\n    at /Users/lukebrown/Documents/dev/cf-graphql/node_modules/express-graphql/dist/index.js:142:52\n    at process._tickCallback (internal/process/next_tick.js:103:7)"
    }
  ],
  "extensions": {
    "time": 3,
    "timeline": []
  }
}
```

This is a copy of content model generated in the Contentful admin of the `category` content type that `service` references 
```javascript
{
  "name": "Category: Type of Business",
  "description": "",
  "displayField": "name",
  "fields": [
    {
      "name": "Name",
      "id": "name",
      "type": "Symbol",
      ...
    },
  ],
  "sys": {
    "id": "category",
    "type": "ContentType",
    ...
  }
}
```

Also interesting is that category.sys works (see below) but apart from `sys` GraphiQL doesn't autocomplete any other fields. Have I misconfigured something? 
```javascript
{
  organisations {
    services {
      category {
	 name // querying this  generates the above error and isn't visible in autocomplete

         // autocompletes and the query works
         sys {
           id 
         }
      }
    } 
  }
}
```

**Comment by jelz** [comment]
Now I get it. I think there is a bug but it doesn't seem to be related to the `include` parameter.

Can you tell me what the following query returns?

```
{
  organisations {
    services {
      category {
         __typename
      }
    }
  }
}
```

**Comment by lkbr** [comment]
This:

```javascript
{
  "data": {
    "organisations": [
      {
        "services": [
          {
            "category": {
              "__typename": "CategoryTypeOfBusiness"
            }
          }
        ]
      }
    ]
  },
  "extensions": {
    "time": 113,
    "timeline": [
      {
        "url": "/entries?content_type=organisation&include=1&limit=100&skip=0",
        "start": 2,
        "duration": 70
      },
      {
        "url": "/entries?include=1&limit=100&skip=0&sys.id%5Bin%5D=48AgUpuPhSWSu0QCCQkeg",
        "start": 75,
        "duration": 37
      }
    ]
  }
}

```

**Comment by jelz** [comment]
Does this query return data you expect?

```
{
  organisations {
    services {
      category {
         ... on CategoryTypeOfBusiness { name }
      }
    }
  }
}
```

**Comment by lkbr** [comment]
Yes it does. I misunderstood the documentation. In the example that demonstrates the `... on` syntax, I considered the `author` and not the `related` example applicable to my data.

Thanks you for the help.

```js
{
  "data": {
    "organisations": [
      {
        "services": [
          {
            "category": {
              "name": "Engineering services"
            }
          }
        ]
      }
    ]
  }
}
```

**Comment by jelz** [comment]
No, it's a bug, there shouldn't be a need of `... on` in your case. You should be able to use query you've provided in your first message.

I'm going to prepare a fix soon.

**Comment by jelz** [comment]
I've just released v0.3 that includes a fix for this issue. Could you check if it solves the problem?

Please note that there were some changes in the way you should run the example (you need to provide a CMA token now).

**Comment by lkbr** [comment]
I've tested this and can confirm its working as expected.

Many thanks.
